model_compression_toolkit/core/common/quantization/quantization_params_generation/power_of_two_selection.py

```diff
@@ -0,0 +1,167 @@
+                                                    channel_axis=channel_axis,
+                                                    n_iter=n_iter,
+                                                    min_threshold=min_threshold)
+        return {THRESHOLD: threshold}
+
+
+def power_of_two_selection_histogram(bins: np.ndarray,
+                                     counts: np.ndarray,
+                                     p: int,
+                                     n_bits: int,
+                                     min_value: float,
+                                     max_value: float,
+                                     constrained: bool = True,
+                                     n_iter: int = 20,
+                                     min_threshold: float = MIN_THRESHOLD,
+                                     quant_error_method: qc.QuantizationErrorMethod = qc.QuantizationErrorMethod.MSE) -> dict:
+    """
+    Compute the optimal threshold based on the provided QuantizationErrorMethod to quantize a histogram.
+    Different search is applied, depends on the value of the selected QuantizationErrorMethod.
+
+    Args:
+        bins: Bins values of the histogram.
+        counts: Bins counts of the histogram.
+        p: p-norm to use for the Lp-norm distance (used only for lp threshold selection).
+        n_bits: Number of bits to quantize the tensor.
+        min_value: Min value (not used for this method).
+        max_value: Max value (not used for this method).
+        constrained: Whether the threshold should be constrained or not (not used for this method).
+        n_iter: Number of iteration ot search for the threshold (not used for this method).
+        min_threshold: Minimal threshold to use if threshold is too small (used only for kl threshold selection).
+        quant_error_method: an error function to optimize the parameters' selection accordingly.
+
+    Returns:
+        Optimal threshold to quantize the histogram a power of 2 manner.
+    """
+    if quant_error_method == qc.QuantizationErrorMethod.NOCLIPPING:
+        tensor_max = np.max(np.abs(bins))
+        threshold = power_of_two_constraint(tensor_max, min_threshold)
+    else:
+        error_function = get_threshold_selection_histogram_error_function(quant_error_method, p)
+        threshold = qparams_selection_histogram_search(error_function,
+                                                       bins,
+                                                       counts,
+                                                       n_bits,
+                                                       constrained=constrained,
+                                                       n_iter=n_iter,
+                                                       min_threshold=min_threshold)
+    return {THRESHOLD: threshold}
+
+
+def get_threshold_selection_tensor_error_function(quant_error_method: qc.QuantizationErrorMethod,
```

Comment:
Can we try to eliminate the code duplication we have here and in the same methods in symmetric + uniform selections?
I no that each one is slightly different, but maybe try to work something that would allow to remove the duplication.

Reply:
Don't know simpler/clearer than that.